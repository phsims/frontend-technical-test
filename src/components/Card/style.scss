@import "../../mixins.scss";

.card {
  border: solid 1px #eff1f1;
  display: flex;
  flex-direction: row;
  height: 100%;
  width: 100%;

  &__image-wrapper {
    flex-shrink: 0;
    height: auto;
    line-height: 0;
    max-width: 25%;
    width: 200px;

    &__picture { 
      display: block;


    }

    &__picture__img {
      width: 100%;
    }
  }

  &__content {
    flex-grow: 1;
    padding: 1rem;

    &-title {
      color: #0c121c;
      font-size: 1.6rem;
      font-weight: 700;
      letter-spacing: .08rem;
      line-height: 1.6rem;
      margin: 0 0 .5rem;
      text-transform: uppercase;


    }

    &-price {
      margin: 0 0 1rem;
    }

    &-description {
      margin: 0 ;

    }
  }
}

@include tablet {

  .card {
    flex-direction: column;

    &__image-wrapper {
      height: auto;
      max-width: 100%;
      width: 100%;
    }

    &__content {
      padding-bottom: .5rem;
      text-align: center;

      &-title {
        line-height: 3.5rem;


      }

      &__content__id { 
        border-bottom: 1px solid #000; 
        border-top: 1px solid #000; 
        padding: .25rem 0;
      }

      &-price {
        margin: 0 0 .5rem;
      }
    }
  }
}