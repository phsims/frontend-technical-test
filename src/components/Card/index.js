import React from 'react';
import './style.scss';

export default function Card({ vehicle }) {
    const { id, price, media, description } = vehicle;
    const largeImage = media.find(mediaItem => mediaItem.url.includes('16x9'));
    const smallImage = media.find(mediaItem => mediaItem.url.includes('1x1'));
    const fallbackImage = largeImage || smallImage;

    return (
        <article className="card">
            {fallbackImage && (
                <div className="card__image-wrapper">
                    <picture className="card__image-wrapper__picture">
                        <source media="(max-width: 767px)" srcSet={smallImage.url} className="card__picture-source" />
                        <source media="(min-width: 768px)" srcSet={largeImage.url} className="card__picture-source" />
                        <img src={fallbackImage.url} alt={fallbackImage.name} className="card__image-wrapper__picture__img" />
                    </picture>
                </div>
            )}
            <div className="card__content">
                <h2 className="card__content-title">
                    <span className="card__content-title__id">{id}</span>
                </h2>
                <p className="card__content-price">From {price}</p>
                <p className="card__content-description">{description}</p>
            </div>
        </article>
    );
}