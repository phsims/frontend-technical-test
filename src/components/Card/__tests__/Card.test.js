import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Card from '../index';

const vehicle = {
  id: '1',
  price: '£100',
  media: [
    { url: '/images/1x1/smallImage.jpg', name: 'image' },
    { url: '/images/16x9/largeImage.jpg', name: 'image' },
  ],
  description: 'A lovely vehicle',
};

describe('Card', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Card vehicle={vehicle} />);
    expect(baseElement).toBeTruthy();
  });

  it('should match the snapshot', () => {
    const { baseElement } = render(<Card vehicle={vehicle} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render with image', async () => {
    const { getByAltText } = render(<Card vehicle={vehicle} />);
    const image = await getByAltText('image');
    expect(image).toHaveAttribute('src', vehicle.media[1].url);
  });
  it('should render with price', async () => {
    const { getByText } = render(<Card vehicle={vehicle} />);
    const price = await getByText(/£100/);
    expect(price).toBeInTheDocument();
  });
  it('should render with description', async () => {
    const { getByText } = render(<Card vehicle={vehicle} />);
    const description = await getByText(/A lovely vehicle/);
    expect(description).toBeInTheDocument();
  });
});
