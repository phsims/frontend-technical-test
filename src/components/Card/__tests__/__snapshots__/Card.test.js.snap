// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
<body>
  <div>
    <article
      class="card"
    >
      <div
        class="card__image-wrapper"
      >
        <picture
          class="card__image-wrapper__picture"
        >
          <source
            class="card__picture-source"
            media="(max-width: 767px)"
            srcset="/images/1x1/smallImage.jpg"
          />
          <source
            class="card__picture-source"
            media="(min-width: 768px)"
            srcset="/images/16x9/largeImage.jpg"
          />
          <img
            alt="image"
            class="card__image-wrapper__picture__img"
            src="/images/16x9/largeImage.jpg"
          />
        </picture>
      </div>
      <div
        class="card__content"
      >
        <h2
          class="card__content-title"
        >
          <span
            class="card__content-title__id"
          >
            1
          </span>
        </h2>
        <p
          class="card__content-price"
        >
          From
          Â£100
        </p>
        <p
          class="card__content-description"
        >
          A lovely vehicle
        </p>
      </div>
    </article>
  </div>
</body>
`;
